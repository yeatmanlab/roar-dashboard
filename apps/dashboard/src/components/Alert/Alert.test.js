import { it, describe, expect } from 'vitest';
import { mount } from '@vue/test-utils';
import { Alert, AlertTitle, AlertDescription, ALERT_VARIANTS } from '.';

describe('Alert Component', () => {
  it('renders the default variant', () => {
    const renderedComponent = mount(Alert, {
      slots: {
        default: [
          mount(AlertTitle, {
            slots: {
              default: 'Error Alert',
            },
          }),
          mount(AlertDescription, {
            slots: {
              default: "Whoops, well that wasn't great, now was it?",
            },
          }),
        ],
      },
    });

    expect(renderedComponent.html()).toMatchSnapshot();
  });

  it('renders the error variant', () => {
    const renderedComponent = mount(Alert, {
      props: {
        variant: ALERT_VARIANTS.DESTRUCTIVE,
      },
      slots: {
        default: [
          mount(AlertTitle, {
            slots: {
              default: 'Error Alert',
            },
          }),
          mount(AlertDescription, {
            slots: {
              default: "Whoops, well that wasn't great, now was it?",
            },
          }),
        ],
      },
    });

    expect(renderedComponent.html()).toMatchSnapshot();
  });
});
