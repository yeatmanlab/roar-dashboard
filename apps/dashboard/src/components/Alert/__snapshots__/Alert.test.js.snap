// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Alert Component > renders the default variant 1`] = `
"<div class="relative w-full rounded border border-solid border-gray-300 p-3 [&>svg]:text-gray-950 flex text-left text-gray-800" role="alert">
  <div class="flex-shrink-0 mt-0.5 mr-3"><i class="text-gray-500 pi pi-info-circle"></i></div>
  <div class="flex flex-column">
    <!---->
    <!---->
  </div>
</div>"
`;

exports[`Alert Component > renders the error variant 1`] = `
"<div class="relative w-full rounded border border-solid p-3 flex text-left bg-red-50 border-red-500/50 text-red-500 [&>svg]:text-red-500" role="alert">
  <div class="flex-shrink-0 mt-0.5 mr-3"><i class="pi pi-exclamation-circle"></i></div>
  <div class="flex flex-column">
    <!---->
    <!---->
  </div>
</div>"
`;
